docs/maybe.md

```diff
@@ -0,0 +1,52 @@
+###E247.Fun
+#Maybe&lt;T&gt;
+
+```Maybe<T>``` is used to represent a value that might not be available. While it is an option to use ```null``` for reference types, it isn't explicit in the type system and returning ```null``` can lead to problems when it's not handled properly, using ```Maybe<T>``` instead makes it obvious that the potential lack of a value needs to be handled, and provides some handy methods for dealing with boilerplate based around that.
+
+##Making a Maybe
+To convert a typed value to a ```Maybe``` you can simply call the ```.ToMaybe()``` extension method, this will even handle if that value is ```null``` by creating an empty ```Maybe``` of that type.
+
+##Checking for a Value
+Before using the value in a ```Maybe``` you should first check if it actually contains one. This can be done by testing the boolean value of the ```.HasValue``` property or the ```.Any()``` method. 
+
+It's also possible to match on both potential states of a ```Maybe``` using the ```.Match``` method which takes 2 functions, one for handling each state:
```

Comment:
Probably it's better to say that .Match should be used instead of .HasValue or .Any() whenever reasonable. Saying "It's also possible" may be intended as if the first option is not using .Match.
In general, I think that .Match should be used whenever we transform the value, with or without a side effect, and .HasValue and .Any() may (or may not) be better if we are doing something only for side effect.